sql/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -591,16 +591,19 @@ public Projector visitWindowAgg(WindowAggProjection windowAgg, Context context)
             ctx.add(functionAndInputsEntry.getValue());
 
             FunctionImplementation impl = this.functions.getQualified(functionAndInputsEntry.getKey().info().ident());
-            assert impl instanceof AggregationFunction : "We currently only support aggregation functions as window functions";
-            windowFunctions.add(
-                new AggregateToWindowFunctionAdapter(
-                    ctx.topLevelInputs().toArray(new Input[0]),
-                    (AggregationFunction) impl,
-                    ctx.expressions(),
-                    indexVersionCreated,
-                    bigArrays,
-                    context.ramAccountingContext)
-            );
+            if (impl instanceof AggregationFunction) {
```

Comment:
not sure if I like checking hierarchy vs a type value. 
I can use the `FunctionInfo.Type` instead eg `if (impl.info().type() == FunctionInfo.Type.WINDOW)`.

What is your preference?

Reply:
If a cast follows it makes imho more sense to use instanceof checks - analysis tooling can then verify that the cast is safe.